### PR TITLE
Migrate API serving based on new AffectedCommits entities.

### PR DESCRIPTION
**Pull request by oliverchang**
Also add a missed case of updating the new entries in the App Engine cron handling making OSS-Fuzz vulns public. We will need to do some manual queries to detect mismatching cases and fix them up if necessary.

Part of #951.

### CODE REVIEW

**Comment by oliverchang** [comment]
Note: Do not merge for this week's prod release.

File: gcp/api/server.py [hunk 1]
```diff
@@ -301,13 +308,16 @@ def _clean_purl(purl):
 @ndb.tasklet
 def query_by_commit(commit, to_response=bug_to_response):
   """Query by commit."""
-  query = osv.AffectedCommit.query(osv.AffectedCommit.commit == commit,
-                                   osv.AffectedCommit.public == True)  # pylint: disable=singleton-comparison
+  query = osv.AffectedCommits.query(osv.AffectedCommits.commits == commit)
```
Comment (michaelkedar) [inline gcp/api/server.py:311]: Is it worth automatically verifying with or falling back to the legacy `AffectedCommit` here, or will that cause too much latency?

File: gcp/appengine/handlers.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ def make_affected_commits_public(bug):
   """Make related AffectedCommit entities public."""
```
Comment (michaelkedar) [inline gcp/appengine/handlers.py:59]: Did a quick search and found this: https://github.com/google/osv.dev/blob/0e31a369fb9055dab2ddf97e0ec3ed7ef5978823/docker/worker/worker.py#L189-L200

Does this also need some handling for the new `AffectedCommits`?

**Review by michaelkedar — APPROVED** [review]
LGTM with a few questions

File: gcp/appengine/handlers.py (hunk 2 above)
Comment (oliverchang) [reply gcp/appengine/handlers.py:59]: Good catch. I've fixed this up as well.

File: gcp/api/server.py (hunk 1 above)
Comment (oliverchang) [reply gcp/api/server.py:311]: Good point. It adds a bit of complexity and I'm a bit wary of latency here. I'd like to validate that these indexes are complete a different way.

**Review by another-rex — APPROVED** [review]
LGTM

**Review by andrewpollock — APPROVED** [review]

**Pull request merged**